gh-69639: add mixed-mode rules for complex arithmetic (C-like) (#124829)
python/cpython: Doc/c-api/complex.rst

   representation.

.. c:function:: Py_complex _Py_cr_sum(Py_complex left, double right)

>    Return the sum of complex and real number, using the C :c:type:`Py_complex`

Reviewer: ```suggestion
   Return the sum of a complex and a real number, using the C :c:type:`Py_complex`
```

>    Return the sum of a complex and a real number, using the C :c:type:`Py_complex`

Reviewer: ```suggestion
   Return the sum of a complex number and a real number, using the C :c:type:`Py_complex`
```

I think we should either always include "number" after "complex/real" or not include it at all. We seem to include it so let's make it consistent.

   representation.

   .. versionadded:: 3.14

.. c:function:: Py_complex _Py_c_diff(Py_complex left, Py_complex right)

   Return the difference between two complex numbers, using the C
   :c:type:`Py_complex` representation.

.. c:function:: Py_complex _Py_cr_diff(Py_complex left, double right)

>    Return the difference of complex and real number, using the C

Reviewer: ```suggestion
   Return the difference between a complex number and a real number, using the C
```

   :c:type:`Py_complex` representation.

   .. versionadded:: 3.14

.. c:function:: Py_complex _Py_rc_diff(double left, Py_complex right)

>    Return the difference of real and complex number, using the C

Reviewer: ```suggestion
   Return the difference between a real number and a complex number, using the C
```

   representation.

.. c:function:: Py_complex _Py_cr_prod(Py_complex left, double right)

>    Return the product of complex and real number, using the C

Reviewer: ```suggestion
   Return the product of a complex number and a real number, using the C
```

   :c:data:`errno` to :c:macro:`!EDOM`.

> .. c:function:: Py_complex _Py_cr_quot(Py_complex left, double right)

Reviewer: ```suggestion
.. c:function:: Py_complex _Py_cr_quot(Py_complex dividend, double divisor)
```

>    Return the quotient of complex and real number, using the C

Reviewer: ```suggestion
   Return the quotient of a complex number and a real number, using the C
```

   :c:type:`Py_complex` representation.

   If *divisor* is zero, this method returns zero and sets
   :c:data:`errno` to :c:macro:`!EDOM`.

   .. versionadded:: 3.14

> .. c:function:: Py_complex _Py_rc_quot(double left, Py_complex right)

Reviewer: ```suggestion
.. c:function:: Py_complex _Py_rc_quot(double dividend, Py_complex divisor)
```

>    Return the quotent of real and complex number, using the C

Reviewer: ```suggestion
   Return the quotient of a real number and a complex number, using the C
```
